Move selectors into their own directory

amyanchen opened the pull request:
Previously, selectors lived in the container files they were used in. Now, they all live in the selectors directory, along with their corresponding tests.

amyanchen commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Move selectors into their own directory](https://trello.com/c/IvFg0oN6/72-move-selectors-into-their-own-directory)

Joonpark13 commented on src/selectors/index.js:
```diff
@@ -1,3 +1,5 @@
+import { meetsDuringDow, meetsDuringHour } from 'components/calendar/calendar-helpers';
```
Now that these are being used outside of components, let's put these in `util/time` instead and import them from there.

Joonpark13 commented on src/selectors/index.js:
```diff
@@ -17,3 +19,113 @@ export function scheduledSectionsSelector(state) {
 export function selectedCourseNameSelector(state) {
   return state.getIn(['browse', 'selected', 'course', 'name']);
 }
+
```
Can we move all of the helper functions to the top, then order the selectors below that by top level store fields? Like all the `search` selectors, then all the `browse` selectors, etc.

Joonpark13 replied:
Wouldn't hurt to delineate each of the three using comments either (see action-types.js)

Joonpark13 commented on src/selectors/index.js:
```diff
@@ -1,7 +1,57 @@
+}
+
+/* schedule */
+export function scheduledSectionsSelector(state) {
```
This should be `schedule` not `scheduled`

Joonpark13 commented on src/selectors/index.js:
```diff
@@ -1,7 +1,57 @@
+  );
+}
+
+/* schedule */
```
No need for this to be a multi line comment. Can we make this `// Schedule` instead? (Same with browse and search)

Joonpark13 commented:
Also seeing a console error on load - can you look into that? Make sure you manually load up and test the app for your PRs!

Joonpark13 commented:
The console error on load is gone but now it errors out when you add a class.

Joonpark13 reviewed (changes requested):
Oh I just noticed you have commits under `images` - not sure how you got that in there but don't think we need those.

amyanchen committed:
Move selectors into their own directory

Previously, selectors lived in the container files they were used in.
Now, they all live in the selectors directory, along with their
corresponding tests.
